pkgs/build-support/setup-hooks/make-binary-wrapper.sh

```diff
@@ -33,6 +33,8 @@ assertExecutable() {
 # To troubleshoot a binary wrapper after you compiled it,
 # use the `strings` command or open the binary file in a text editor.
 makeWrapper() {
+    local NIX_CFLAGS_COMPILE NIX_CFLAGS_LINK
+    unset NIX_CFLAGS_COMPILE NIX_CFLAGS_LINK
```

Comment:
Why do we need this?

Reply:
cf point 2 in OP

Reply:
Please put this into the commit message.

Reply:
Sure, I'll make nice commit messages when this is a bit more figured out (I'm still waiting for feedback on whether this is the right way, and what other variables we should unset)